core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -829,6 +829,55 @@ class SparkContext(config: SparkConf) extends Logging {
     setLocalProperty(SparkContext.SPARK_JOB_INTERRUPT_ON_CANCEL, null)
   }
 
+  /**
+   * Set the behavior of job cancellation from jobs started in this thread.
+   *
+   * @param interruptOnCancel If true, then job cancellation will result in `Thread.interrupt()`
+   * being called on the job's executor threads. This is useful to help ensure that the tasks
+   * are actually stopped in a timely manner, but is off by default due to HDFS-1208, where HDFS
```

Comment:
Someone should check if this is still a thing :)

Reply:
The HDFS-1208 bug is still open... but multiple places in core of Spark has by now elected to just pass `true` here, so it likely doesn't make sense for the user to set it to `false`, as these places would generate interrupts anyway...